packages/modular-scripts/src/utils/config.ts

```diff
@@ -0,0 +1,108 @@
+import { cosmiconfigSync } from 'cosmiconfig';
+import path from 'path';
+import getModularRoot from './getModularRoot';
+
+// Where cosmiconfig can look for the configuration
+const searchPlaces = [
+  '.modular.js',
+  'package.json',
+  `.modularrc`,
+  `.modularrc.json`,
+  `.modularrc.yaml`,
+  `.modularrc.yml`,
+  `.modularrc.js`,
+  `.modularrc.cjs`,
+  `modular.config.js`,
+  `modular.config.cjs`,
+];
+
+// Look for configuration file
+const modularRoot = getModularRoot();
+const explorer = cosmiconfigSync('modular', { searchPlaces });
+const configResult = explorer.search(path.join(modularRoot, 'package.json'));
```

Comment:
If you were to cast the type here, you can remove the inline type assertions in the `getConfig` below:

```suggestion
const configResult: undefined | { config: Partial<ConfigObject> }  = explorer.search(path.join(modularRoot, 'package.json'));
```

Using a `Partial<T>` in this type would mean you can remove the `| null` from each property of `ConfigObject` along with all the `NonNullable<T>` inline types

Reply:
I've given it a go, much cleaner now, but doesn't feel quite 'perfect' yet, feels like there should be a better way of doing: 
```ts
  let configValue;
  if (configResult) {
    const loadedConfigValue = configResult.config[key];
    if (typeof loadedConfigValue === typeof defs[key].default) {
      configValue = loadedConfigValue as Exclude<Config[T], null>;
    }
  }
  return defs[key].override ?? configValue ?? defs[key].default;
```

Reply:
Updated attempt:
```js
export function getConfig<T extends keyof ConfigDefs>(key: T): Config[T] {
  const configValue: Config[T] | undefined = configResult
    ? configResult.config[key]
    : undefined;
  return defs[key].override ?? configValue ?? defs[key].default;
}
```

quite happy with that

Reply:
I used the two types in my suggestion, but you should probably collapse to just the one type in the `getConfig` method:

```typescript
export function getConfig<T extends keyof Config>(key: T): Config[T] {
```